test/widgets/message_list_test.dart

```diff
@@ -51,6 +59,58 @@ Future<void> setupMessageListPage(WidgetTester tester, {
 void main() {
   TestZulipBinding.ensureInitialized();
 
+  group('MessageWithSender', () {
+    testWidgets('Updates avatar on RealmUserUpdateEvent', (tester) async {
+      addTearDown(TestZulipBinding.instance.reset);
+
+      RealmContentNetworkImage? findAvatarImageWidget(WidgetTester tester) {
+        final firstMessageWithSender = tester.widgetList(find.byType(MessageWithSender)).first;
+        return tester.widgetList<RealmContentNetworkImage>(
+          find.descendant(
+            of: find.byWidget(firstMessageWithSender),
```

Comment:
Whichever way it's organized, I don't love that we have to look for RealmContentNetworkImage here to find the avatar. That feels like a bit of an implementation detail of the message list, and also a bit fragile because that widget type could appear in the content. I'd rather have something we can look for that more specifically marks the avatar.

I guess with an `Avatar` widget that'd provide an answer. Though that also feels like a bit of an implementation detail.

Another version would be to look for a semantic label. That'd be appealing because it's not an implementation detail — it's something user-facing. Not sure what the right semantic label would be in this case, though. In fact I kind of think for the semantics tree we likely want to *hide* the avatar — there's nothing it communicates that isn't already covered by the sender name, other than the visual of the image itself.

Anyway, using RealmContentNetworkImage is fine for now. We can switch to the Avatar widget upon adding that, and possibly to semantics later.

Reply:
Makes sense. Yeah, I felt kind of uneasy having it look for `RealmContentNetworkImage` too.

Reply:
> Also this could just be the `find.byType(MessageWithSender)` directly, and then `firstWidget` around the whole thing.

Ah: I tried this just now, but then a test failure reminded me that we'd need it to be firstWidget*OrNull*, and that isn't offered.